### PR TITLE
Implement new keycard factory reset flow

### PR DESCRIPTION
**Pull request by saledjenic**
This partially covers factory reset flow. The part where user is able to select which accounts wants to remove/keep from/on a keycard will be added later, once we add the keycard settings part for storing those data to a keycard.

Second part of factory reset flow will be developed separately, logged issue https://github.com/status-im/status-desktop/issues/6922

Corresponding `StatusQ` PR:
- https://github.com/status-im/StatusQ/pull/845

Closes: #6790

### CODE REVIEW

File: ui/imports/shared/popups/keycard/states/KeycardInit.qml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        }
+    }
+
+    states: [
+        State {
+            name: Constants.keycardSharedState.pluginReader
+            when: root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.pluginReader
+            PropertyChanges {
+                target: title
+                text: qsTrId("Plug in Keycard reader...")
```
Comment (jrainville) [inline ui/imports/shared/popups/keycard/states/KeycardInit.qml:75]:
```suggestion
                text: qsTr("Plug in Keycard reader...")
```

File: ui/imports/shared/popups/keycard/states/KeycardInit.qml [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                text: ""
+                visible: false
+            }
+        },
+        State {
+            name: Constants.keycardSharedState.insertKeycard
+            when: root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.insertKeycard
+            PropertyChanges {
+                target: title
+                text: qsTrId("Insert Keycard...")
```
Comment (jrainville) [inline ui/imports/shared/popups/keycard/states/KeycardInit.qml:95]:
```suggestion
                text: qsTr("Insert Keycard...")
```

**Review by jrainville — COMMENTED** [review]
Good job, just some small comments

File: src/app/core/eventemitter.nim [hunk 3]
```diff
@@ -23,15 +23,26 @@ proc on(this: EventEmitter, name: string, handlerId: UUID, handler: Handler): vo
...
+  var handlerId = genUUID()
+  this.on(name, handlerId, handler)
 
 proc once*(this:EventEmitter, name:string, handler:Handler): void =
   var handlerId = genUUID()
   this.on(name, handlerId) do(a: Args):
     handler(a)
     this.events[name].del handlerId
 
+proc onWithUUID*(this: EventEmitter, name: string, handler: Handler): UUID =
```
[the remark this answers is not in the document]
Comment (saledjenic) [reply src/app/core/eventemitter.nim:35]: @jrainville yes, that was the main thing why I didn't update `on`, I would need to discard thousand places where we're using it...

**Comment by saledjenic** [comment]
@jrainville thanks for the comments, all fixed.

**Review by jrainville — APPROVED** [review]

File: ui/imports/shared/popups/keycard/KeycardPopup.qml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+                onConfirmationUpdated: {
+                    d.factoryResetConfirmed = value
+                }
+            }
+        }
+    }
+
+    leftButtons: [
+        StatusRoundButton {
```
Comment (caybro) [inline ui/imports/shared/popups/keycard/KeycardPopup.qml:91]: we have StatusBackButton now

File: ui/imports/shared/popups/keycard/states/KeycardConfirmation.qml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            mipmap: true
+            source: Style.png("keycard/popup_card_red_sprayed@2x")
+        }
+
+        StatusBaseText {
+            id: title
+            Layout.alignment: Qt.AlignHCenter
+            horizontalAlignment: Text.AlignHCenter
+            wrapMode: Text.WordWrap
+            text: qsTr(" A factory reset will delete the key on this Keycard.\nAre you sure you want to do this?")
```
Comment (caybro) [inline ui/imports/shared/popups/keycard/states/KeycardConfirmation.qml:38]: Stray space at the start?

File: ui/imports/shared/popups/keycard/states/KeycardInit.qml [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            Layout.preferredHeight: Constants.keycard.shared.imageHeight
+            Layout.preferredWidth: Constants.keycard.shared.imageWidth
+            fillMode: Image.PreserveAspectFit
+            antialiasing: true
+            mipmap: true
+        }
+
+        Row {
+            spacing: Style.current.halfPadding
+            Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
```
Comment (caybro) [inline ui/imports/shared/popups/keycard/states/KeycardInit.qml:41]: => `Qt.AlignCenter`

**Review by caybro — APPROVED** [review]
Just some minor suggestions, looks cool though

**Comment by anastasiyaig** [comment]
@saledjenic please rebase

**Comment by saledjenic** [comment]
@caybro thanks for the comments, all updated.

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**